cogs/events.py

```diff
@@ -1,21 +1,54 @@
 import traceback
 
-from discord import Color, Embed, Message
+from discord import Color, Embed, Message, Guild, Member
 from discord.ext.commands import Bot, Cog
 
+import re
+import aiohttp
+
 
 class Custom(Cog):
     def __init__(self, bot: Bot) -> None:
         self.bot = bot
 
+    @staticmethod
+    def get_link_code(string: str) -> str:
+        return string.split("/")[-1]
+
+    @staticmethod
+    async def check_our_invite(full_link: str, guild: Guild) -> bool:
+        guild_invites = await guild.invites()
+        for invite in guild_invites:
+            # discord.gg/code resolves to https://discordapp.com/invite/code after using session.get(invite)
+            if Custom.get_link_code(invite.url) == Custom.get_link_code(full_link):
+                return True
+        return False
+
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         # Check that the message is not from bot account
         if message.author.bot:
             return
 
-        if message.content.lower().startswith("help"):
-            await message.channel.send("Hey! Why don't you run the help command with `>>help`")
+        elif not isinstance(message.author, Member):
+            return
```

Comment:
When does this happen? I assume it's a check for DMs, because I don't see `message.author` not being `discord.Member` in any other way, not even sure about the DMs? If this is a DM check it should be commented above.